test/launcher_test.rb

```diff
@@ -0,0 +1,36 @@
+require 'test_helper'
+require 'launcher'
+
+class LauncherTest < Test::Unit::TestCase
+  def setup
+    @launcher = Proxy::Launcher.new
+    @launcher.stubs(:pid_path).returns("launcher_test.pid")
+  end
+
+  def test_pid_status_exited
+    assert_equal :exited, @launcher.pid_status
+  end
+
+  def test_write_pid_success
+    @launcher.write_pid
+    assert File.exist?(@launcher.pid_path)
+  ensure
+    FileUtils.rm_f @launcher.pid_path
+  end
+
+  def test_pid_status_running
+    @launcher.write_pid
+    assert_equal :running, @launcher.pid_status
+  ensure
+    FileUtils.rm_f @launcher.pid_path
+  end
+
+  def test_write_pid_errj
```

Comment:
> I think the name is misleading -- you are testing if check_pid forces exit if status is `:running`
>

This was a typo. Corrected to `test_check_pid_exits_program`

-- 
Later,
 Lukas #lzap Zapletal